app/AppKernel.php

```diff
@@ -175,6 +181,53 @@ private function enableComposerAutoloaderOnModules($modules)
         }
     }
 
+
+    /**
+     * Add Bundles from active Modules to the collection.
+     */
+    private function loadModulesBundles(array &$bundles)
+    {
+        $activeModules = $this->getActiveModules();
+        /** @var SplFileInfo $moduleFolder */
+        foreach ($this->getModulesFolders() as $moduleFolder) {
+            if (in_array($moduleFolder->getFilename(), $activeModules)) {
+                $this->loadModuleBundle($moduleFolder, $bundles);
+            }
+        }
+    }
+
+    /**
+     * Checks if a single module has bundle file and add it to the collection.
+     */
+    private function loadModuleBundle(SplFileInfo $moduleFolder, array &$bundles)
+    {
+        $moduleSrcDirectory = $moduleFolder->getRealPath() . DIRECTORY_SEPARATOR . 'src';
+        if (!is_dir($moduleSrcDirectory)) {
+            return;
+        }
+
+        $finder = new Finder();
+        $finder->files()
+            ->in($moduleSrcDirectory)
+            ->depth(0)
+            ->name('*Bundle.php')
+        ;
+        foreach ($finder as $phpFile) {
+            $className = PhpFileParser::getCLassName($phpFile);
+            if (is_a($className, Bundle::class, true)) {
```

Comment:
Prefer instanceof, it's an instruction which is faster
```suggestion
            if ($className instanceof Bundle) {
```

Reply:
instanceof requires to have an instance. I did not want to instanciate the class only to make sure it is a bundle or not

Reply:
I dont know if `is_a` do instanciate the class behind the scene

Reply:
It doesn't call the constructor: https://3v4l.org/RUWpH

Reply:
You're right, forget the class isn't instantiated yet :sweat_smile: